ruby_basics/lib/holes.rb

```diff
@@ -0,0 +1,33 @@
+# Implement a function that counts the number of holes(enclosed spaces) in a number. Use the
+# following table as a reference for the number of holes of a specific digit. For example the input
+# number 100 should return a result of 2
+#
+# Digit | Holes
+# ------+------
+#   0   |   1
+#   1   |   0
+#   2   |   0
+#   3   |   0
+#   4   |   1
+#   5   |   0
+#   6   |   1
+#   7   |   0
+#   8   |   2
+#   9   |   1
+
+def holes(number) # rubocop:disable Metrics/MethodLength
+  number_of_holes = {
+    0 => 1,
+    1 => 0,
+    2 => 0,
+    3 => 0,
+    4 => 1,
+    5 => 0,
+    6 => 1,
+    7 => 0,
+    8 => 2,
+    9 => 1
+  }
+
+  number.digits.sum { |digit| number_of_holes[digit] }
```

Comment:
Ova linija takodjer moze biti napisana kao:
```ruby
number.digits.sum(&number_of_holes)
```

I vjerovao ili ne, shvatiti zasto to tocno radi je zanimljiva i korisna vjezba. Tako da ako ti nije vec jasno, ti to ostavaljam kao mini zagonetku. Objasnim uzivo ako sam ne skuzis do onda. :)